### PR TITLE
refactor: components ClipboardButton and ClipboardIcon

### LINKED ISSUE #1019: Refactor ClipboardButton
1. Get rid off Button's private CSS variables usage
2. Update icons #866

### PR DESCRIPTION
**Pull request by Raubzeug**
Closes #1019

### CODE REVIEW

**Review by korvin89 — APPROVED** [review]

File: src/components/ClipboardButton/README.md [hunk 1]
```diff
@@ -30,18 +30,40 @@ LANDING_BLOCK-->
...
-| hasTooltip         | Disable tooltip. Tooltip won't be shown                                  |                     `boolean`                     |   `true`    |
+| text               | Text to copy                                                             |                     `string`                      |             |
+| timeout            | Time before state bounces back to its normal after the button is clicked |                     `number`                      |   `1000`    |
 | tooltipInitialText | Text shown before copy                                                   |                     `string`                      |  `"Copy"`   |
 | tooltipSuccessText | Text shown after copy                                                    |                     `string`                      | `"Copied!"` |
-| qa                 | HTML `data-qa` attribute, used in tests                                  |                     `string`                      |             |
-| timeout            | Time before state bounces back to its normal after the button is clicked |                     `number`                      |   `1000`    |
+
+##### Properties inherited from Button
```
Comment (amje) [inline src/components/ClipboardButton/README.md:45]: No need to duplicate them, let's put a reference to the Button's readme

File: src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import './ClipboardIcon.stories.scss';
+
+export default {
+    title: 'Components/Utils/ClipboardIcon',
+    component: ClipboardIcon,
+    args: {size: 24},
+} as Meta;
+
+const DefaultTemplate: StoryFn<ClipboardIconProps> = (args) => (
```
Comment (amje) [inline src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx:16]: Let's use new stories definition. Look at Button for reference

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <div className="clipboard-icon-stories">
+        Default: <ClipboardIcon {...args} status="pending" />
+        Success: <ClipboardIcon {...args} status="success" />
+        Error: <ClipboardIcon {...args} status="error" />
+    </div>
+);
+export const Default = DefaultTemplate.bind({});
```
Comment (amje) [inline src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx:23]: Let's add 3 separate stories: Default, Success, Error

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/ClipboardButton/ClipboardButton.tsx [hunk 4]
```diff
@@ -37,44 +27,36 @@ interface ClipboardButtonComponentProps extends QAProps {
...
-
-const DEFAULT_ICON_SIZE = 24;
 const DEFAULT_TIMEOUT = 1000;
 
+const ButtonSizeToIconSize: Record<ButtonSize, number> = {
+    xs: 14,
+    s: 18,
+    m: 22,
+    l: 30,
+    xl: 38,
```
Comment (amje) [inline src/components/ClipboardButton/ClipboardButton.tsx:37]: Aren't these sizes too large?

File: src/components/ClipboardButton/README.md (hunk 1 above)
Comment (Raubzeug) [reply src/components/ClipboardButton/README.md:45]: I've thought about it, but in ClipboardButton we don't have a full range of Button props. How should we mention this?..

File: src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx [hunk 5]
```diff
@@ -1,23 +1,31 @@
...
+
+export const Default: Story = {args: {status: 'pending'}};
+
+export const Success: Story = {
+    render: (args) => (
+        <Showcase>
+            <ClipboardIcon {...args} status="success" />
+        </Showcase>
+    ),
+};
```
Comment (amje) [inline src/components/ClipboardIcon/__stories__/ClipboardIcon.stories.tsx:24]: No need to wrap in Showcase for single element story
```tsx
export const Success: Story = {args: {status: 'success'}};
```

And same for Error

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/ClipboardButton/ClipboardButton.tsx (hunk 4 above)
Comment (Raubzeug) [reply src/components/ClipboardButton/ClipboardButton.tsx]: Agreed to use smaller icon sizes.

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**